File: Python/205.py
class Solution(object):
    def isIsomorphic(self, s, t):
        d = {}
        for i in range(len(s)):
            if s[i] not in d:
                d[s[i]] = t[i]
            else:
                if d[s[i]] != t[i]:
                    return False
        d = {}
        for i in range(len(s)):
            if t[i] not in d:
                d[t[i]] = s[i]
            else:
                if d[t[i]] != s[i]:
                    return False
        return True

File: Java/205.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        char[] a = new char[s.length()];
        char[] b = new char[t.length()];
        int k = 0;
        for(int i=0;i<s.length();i++){
            int flag = 0;
            for(int j=0;j<k;j++){
                if(s.charAt(i) == a[j]){
                    if(t.charAt(i) == b[j]){
                        flag = 1;
                    }
                    else{
                        return false;
                    }
                }
                if(t.charAt(i) == b[j]){
                    if(s.charAt(i) == a[j]){
                        flag = 1;
                    }
                    else{
                        return false;
                    }
                }
            }
            if(flag == 0){
                a[k] = s.charAt(i);
                b[k] = t.charAt(i);
                k++; 
            }
        }
        return true;
    }
}